integration-tests/bats/pull.bats

```diff
@@ -0,0 +1,378 @@
+#!/usr/bin/env bats
+load $BATS_TEST_DIRNAME/helper/common.bash
+
+setup() {
+    setup_common
+
+    TMPDIRS=$(pwd)/tmpdirs
+    mkdir -p $TMPDIRS/{rem1,repo1}
+
+    # repo1 -> rem1 -> repo2
+    cd $TMPDIRS/repo1
+    dolt init
+    dolt branch feature
+    dolt remote add origin file://../rem1
+    dolt remote add test-remote file://../rem1
+    dolt push origin main
+
+    cd $TMPDIRS
+    dolt clone file://rem1 repo2
+    cd $TMPDIRS/repo2
+    dolt log
+    dolt branch feature
+    dolt remote add test-remote file://../rem1
+
+    # table and commits only present on repo1, rem1 at start
+    cd $TMPDIRS/repo1
+    dolt sql -q "create table t1 (a int primary key, b int)"
+    dolt add .
+    dolt commit -am "First commit"
+    dolt sql -q "insert into t1 values (0,0)"
+    dolt commit -am "Second commit"
+    dolt push origin main
+    cd $TMPDIRS
+}
+
+teardown() {
+    teardown_common
+    rm -rf $TMPDIRS
+    cd $BATS_TMPDIR
+}
+
+@test "pull: pull main" {
+    cd repo2
+    dolt pull origin
+    run dolt sql -q "show tables" -r csv
+    [ "$status" -eq 0 ]
+    [ "${#lines[@]}" -eq 2 ]
+    [[ "$output" =~ "Table" ]] || false
+    [[ "$output" =~ "t1" ]] || false
```

Comment:
My feedback applies to pretty much every test here - there isn't any validation that the HEAD actually changed as a result of the merge. It's not sufficient to look at the output text and status. Is there a common pattern you could apply to all tests which verifies that the content of each parent in the merge is in the final result?